multi-node/aws/pkg/cluster/cluster.go

```diff
@@ -302,18 +302,26 @@ func (c *Cluster) validateDNSConfig(r53 r53Service) error {
 		return nil
 	}
 
-	zonesResp, err := r53.ListHostedZonesByName(&route53.ListHostedZonesByNameInput{
+	listHostedZoneInput := route53.ListHostedZonesByNameInput{
 		DNSName: aws.String(c.HostedZone),
-	})
+	}
+	if c.HostedZoneID != "" {
+		listHostedZoneInput.HostedZoneId = aws.String(c.HostedZoneID)
+	}
+
+	zonesResp, err := r53.ListHostedZonesByName(&listHostedZoneInput)
 	if err != nil {
 		return fmt.Errorf("Error validating HostedZone: %s", err)
 	}
 
 	zones := zonesResp.HostedZones
-	if len(zones) == 0 || (*zones[0].Name != c.HostedZone) {
+
+	zoneNameMismatch := func() bool { return *zones[0].Name != c.HostedZone }
```

Comment:
The trailing dot should always be there on `c.HostedZone`, it gets added in `config.ClusterFromBytes`